assets/js/components/StatesBarChart.vue

```diff
@@ -365,17 +365,52 @@ export default Chart.extend({
         }, {})
     },
 
+    /**
+     * returns projects/programmes for donors/beneficiaries
+     * if a donor or beneficiary is selected (filter is applied)
+     * it will show the projects/programmes for that state
+     */
     legend_items() {
       // just the totals merged into the types
       const out = {}
+
       for (const t in this.types) {
-        out[t] = Object.assign({
-          value: this.totals[t],
-        }, this.types[t])
+        if(this.filters[this.state_type]) {
```

Comment:
this would be so much more readable with an inline conditional:
```js
Object.assign({
  value: condition ? that_thing : the_other_thing,
}, ...
```
(which happens to be a one-line diff...)

also, all that comment above the function could be placed above the "value: ..." line, and summarized to, say, "return state values when filtered by state, or aggregated totals otherwise"